src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+public class Main {
+    private static final int ARRAY_SIZE = 6;
+    private static final FigureSupplier figureSupplier = new FigureSupplier();
+
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[ARRAY_SIZE];
+        for (int i = 0; i < figures.length; i++) {
+            if (i < figures.length / 2) {
+                figures[i] = figureSupplier.getRandomFigure();
+            } else {
+                figures[i] = figureSupplier.getDefaultFigure();
+            }
+            figures[i].drawFigure();
```

Comment:
The 'drawFigure()' method should not be called inside the loop that is filling the array. Separate the concerns by first filling the array with figures and then iterate over the array in a separate loop to draw the figures.